packages/svelte/src/internal/client/runtime.js

```diff
@@ -588,7 +589,7 @@ export function schedule_effect(signal) {
 
 		if ((flags & BRANCH_EFFECT) !== 0) {
 			if ((flags & CLEAN) === 0) return;
-			set_signal_status(effect, MAYBE_DIRTY);
+			set_signal_status(effect, EFFECT_HAS_DIRTY_CHILDREN);
```

Comment:
is this actually doing anything? `EFFECT_HAS_DIRTY_CHILDREN` is never unset (because `STATUS_MASK` is unaware of it); if the purpose is just to unset `CLEAN` than `MAYBE_DIRTY` or even `0` works just as well

Reply:
I updated the code to include `EFFECT_HAS_DIRTY_CHILDREN` in the logic.

Reply:
`set_signal_status(effect, 0)` works great — `set_signal_status` removes any existing status flags and applies the provided one, and since all we care about is unsetting `CLEAN` we don't _need_ to provide one.

But since all we care about is unsetting `CLEAN`, we don't even need to do that, we can just do this:

```suggestion
			effect.f ^= CLEAN;
```

In other words if we're not going to separate _status_ from _has dirty children_ then `EFFECT_HAS_DIRTY_CHILDREN` is surplus to requirements. I still think that the code would be more self-explanatory that way, and perhaps there's some synthesis of this PR and #13297 that could get us there, but in the meantime I'll go ahead and make that change